src/loaders/resourceLoader.ts

```diff
@@ -493,3 +492,219 @@ export abstract class ResourceLoader implements IResourceBase {
     return schemaRegistry;
   }
 }
+
+export abstract class CachingResourceLoader<
+  ET extends EnvironmentType,
+  KCT extends KafkaClusterType,
+  SRT extends SchemaRegistryType,
+> extends ResourceLoader {
```

Comment:
Minor: any reason this shouldn't go in its own module?

Reply:
Moved.